File: shared/src/main/java/com/emanuelvini/nuven/core/shared/logger/PlatformLogger.java
package com.emanuelvini.nuven.core.shared.logger;

public interface PlatformLogger {

    void log(String... messages);

    default void success(String message) {
        log("§a" + message);
    }
    default void error(String message) {
        log("§c" + message);
    }
    default void warn(String message) {
        log("§e" + message);
    }

}

File: shared/src/main/kotlin/com/emanuelvini/nuven/core/shared/logger/PlatformLogger.kt
package com.emanuelvini.nuven.core.shared.logger

interface PlatformLogger {
    fun log(vararg messages: String?)
    fun success(message: String) {
        log("§a$message")
    }

    fun error(message: String) {
        log("§c$message")
    }

    fun warn(message: String) {
        log("§e$message")
    }
}
